packages/editor/src/components/post-sync-status/index.js

```diff
@@ -1,34 +1,110 @@
 /**
  * WordPress dependencies
  */
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { PanelRow } from '@wordpress/components';
+import {
+	PanelRow,
+	Modal,
+	Button,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	ToggleControl,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { ReusableBlocksRenameHint } from '@wordpress/block-editor';
+import { getQueryArgs } from '@wordpress/url';
 
 /**
  * Internal dependencies
  */
 import { store as editorStore } from '../../store';
 
 export default function PostSyncStatus() {
-	const { syncStatus, postType } = useSelect( ( select ) => {
-		const { getEditedPostAttribute } = select( editorStore );
+	const { editPost } = useDispatch( editorStore );
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const [ syncType, setSyncType ] = useState( 'fully' );
+
+	const { syncStatus, postType, isNewPost } = useSelect( ( select ) => {
+		const { getEditedPostAttribute, isCleanNewPost } =
+			select( editorStore );
 		return {
 			syncStatus: getEditedPostAttribute( 'wp_pattern_sync_status' ),
 			postType: getEditedPostAttribute( 'type' ),
+			isNewPost: isCleanNewPost(),
 		};
 	}, [] );
+
+	useEffect( () => {
+		if ( isNewPost ) {
+			setIsModalOpen( true );
+		}
+		// We only want the modal to open when the page is first loaded.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
+
+	const setSyncStatus = () => {
+		editPost( {
+			meta: {
+				wp_pattern_sync_status:
+					syncType === 'unsynced' ? 'unsynced' : null,
```

Comment:
This isn't working quite right for me, and I'm not sure it's doing anything.

I opt into creating a an unsynced pattern, then publish it. But when I use it in a post it is synced.

In fact all patterns created in this flow are synced by default with or without this callback. I assume it's a default value of the `wp_pattern_sync_status` property of the `WP_Block` class/type.

Therefore `wp_pattern_sync_status` is always `null` despite this `editPost()` call - and possibly why I'm seeing an [error](https://github.com/WordPress/gutenberg/pull/52352#issuecomment-1628368375) when I try to publish a synced pattern? That is, setting `null` to what's already `null`??

I don't know.

Reply:
should have been set to `undefined` instead of `null`